PASS: tests/errors/syn7.kt
PASS: tests/errors/lex_test42.kt
PASS: tests/errors/lex_test12.kt
PASS: tests/errors/syn2.kt
PASS: tests/errors/syn3.kt
PASS: tests/errors/lex_test53.kt
FAIL: tests/errors/syn8.kt
Expected Error, but got:
Processing file: tests/errors/syn8.kt
DEBUG: Function 'outer' recorded with 0 parameter(s), return type: any
DEBUG: Function 'inner' recorded with 0 parameter(s), return type: any
DEBUG: Entering node: functionDeclaration (prod: 327) at line 5, scope: 0x55bd1259eed0
DEBUG: Function declaration encountered. Updating current scope.
DEBUG: Function 'outer' has scope pointer: 0x55bd1259ebd0
DEBUG: Entering node: outer (prod: 396) at line 0, scope: 0x55bd1259ebd0
DEBUG: Entering node: functionValueParameters (prod: 108) at line 1, scope: 0x55bd1259ebd0
DEBUG: Entering node: block (prod: 111) at line 5, scope: 0x55bd1259ebd0
DEBUG: Entering node: functionDeclaration (prod: 327) at line 4, scope: 0x55bd1259ebd0
DEBUG: Function declaration encountered. Updating current scope.
DEBUG: Function 'inner' has scope pointer: 0x55bd1259eed0
DEBUG: Entering node: inner (prod: 396) at line 0, scope: 0x55bd1259eed0
DEBUG: Entering node: functionValueParameters (prod: 108) at line 2, scope: 0x55bd1259eed0
DEBUG: Entering node: block (prod: 111) at line 4, scope: 0x55bd1259eed0
DEBUG: Entering node: functionCall (prod: 122) at line 4, scope: 0x55bd1259eed0
DEBUG: Entering node: println (prod: 396) at line 0, scope: 0x55bd1259eed0
DEBUG: Entering node: expressionList (prod: 116) at line 3, scope: 0x55bd1259eed0
DEBUG: Entering node: StringLiteral (prod: 395) at line 3, scope: 0x55bd1259eed0
DEBUG: Entering node: "Nested function" (prod: 395) at line 0, scope: 0x55bd1259eed0
DEBUG: Checking function call for 'println' at line 0
DEBUG: Function 'println' expects 1 parameter(s), call has 1 argument(s).
No errors

PASS: tests/errors/lex_test39.kt
PASS: tests/errors/sem7.kt
PASS: tests/errors/syn_test2.kt
PASS: tests/errors/lex_test19.kt
PASS: tests/errors/lex_test27.kt
PASS: tests/errors/lex_test25.kt
PASS: tests/errors/lex_test54.kt
PASS: tests/errors/lex_test4.kt
PASS: tests/errors/lex_test11.kt
PASS: tests/errors/lex_test21.kt
PASS: tests/errors/sem4.kt
PASS: tests/errors/sem8.kt
PASS: tests/errors/lex_test13.kt
PASS: tests/errors/lex_test1.kt
PASS: tests/errors/lex_test22.kt
PASS: tests/errors/lex_test26.kt
PASS: tests/errors/lex_test24.kt
PASS: tests/errors/sem3.kt
PASS: tests/errors/lex_test51.kt
PASS: tests/errors/lex.kt
PASS: tests/errors/lex_test18.kt
PASS: tests/errors/lex_test0.kt
PASS: tests/errors/lex_test46.kt
PASS: tests/errors/lex_test36.kt
PASS: tests/errors/lex_test47.kt
PASS: tests/errors/sem.kt
PASS: tests/errors/lex_test49.kt
PASS: tests/errors/sem_test4.kt
PASS: tests/errors/lex_test45.kt
PASS: tests/errors/lex_test20.kt
PASS: tests/errors/lex_test23.kt
PASS: tests/errors/lex_test48.kt
PASS: tests/errors/lex_test32.kt
PASS: tests/errors/sem9.kt
PASS: tests/errors/lex_test37.kt
PASS: tests/errors/lex_test29.kt
PASS: tests/errors/syn4.kt
PASS: tests/errors/lex_test33.kt
PASS: tests/errors/syn5.kt
PASS: tests/errors/lex_test16.kt
PASS: tests/errors/lex_test52.kt
PASS: tests/errors/lex_test6.kt
PASS: tests/errors/lex_test7.kt
PASS: tests/errors/sem10.kt
PASS: tests/errors/lex_test40.kt
PASS: tests/errors/lex_test28.kt
PASS: tests/errors/lex_test8.kt
PASS: tests/errors/sem6.kt
PASS: tests/errors/lex_test55.kt
PASS: tests/errors/lex_test30.kt
PASS: tests/errors/sem5.kt
PASS: tests/errors/lex_test14.kt
PASS: tests/errors/lex_test31.kt
PASS: tests/errors/lex_test17.kt
PASS: tests/errors/syn.kt
PASS: tests/errors/lex_test50.kt
PASS: tests/errors/lex_test3.kt
PASS: tests/errors/lex_test43.kt
PASS: tests/errors/lex_test44.kt
PASS: tests/errors/lex_test9.kt
PASS: tests/errors/lex_test2.kt
PASS: tests/errors/lex_test35.kt
PASS: tests/errors/lex_test34.kt
PASS: tests/errors/sem2.kt
PASS: tests/errors/lex_test38.kt
PASS: tests/errors/lex_test10.kt
PASS: tests/errors/lex_test41.kt
PASS: tests/errors/lex_test15.kt
PASS: tests/errors/lex_test5.kt
PASS: tests/k0/syn6.kt
PASS: tests/k0/syn1.kt
PASS: tests/k0/syn_hellowrld.kt
FAIL: tests/k0/sem1.kt
Expected No Error, but got:
Semantic Error at line 0: Undefined function
Semantic Error at line 0: Undefined function
Semantic Error at line 0: Undefined function
Semantic Error at line 0: Undefined function
Semantic Error at line 0: Undefined function
Semantic Error at line 0: Undefined function
Semantic Error at line 80: Function call argument type mismatch for parameter 0: expected any, got none
Semantic Error at line 0: Undefined function
Semantic Error at line 81: Function call argument type mismatch for parameter 0: expected any, got none
Semantic Error at line 0: Undefined function
Semantic Error at line 82: Function call argument type mismatch for parameter 0: expected any, got none
Semantic Error at line 0: Undefined function
Semantic Error at line 83: Function call argument type mismatch for parameter 0: expected any, got none
Semantic Error at line 0: Undefined function
Semantic Error at line 84: Function call argument type mismatch for parameter 0: expected any, got none
Semantic Error at line 0: Undefined function
Semantic Error at line 85: Function call argument type mismatch for parameter 0: expected any, got none
Semantic Error at line 0: Undefined function
Semantic Error at line 86: Function call argument type mismatch for parameter 0: expected any, got none
Semantic Error at line 0: Undefined function
Semantic Error at line 87: Function call argument type mismatch for parameter 0: expected any, got none

Parsing completed with 21 semantic error(s)

PASS: tests/k0/syn_multiply.kt
FAIL: tests/k0/lex_test1.kt
Expected No Error, but got:

Error #1 at line 76: syntax error
Near token: '
'
Line 76:     val isGreaterThan: Boolean = constantValue > mutableVariable
        ^
Hint: Check for missing semicolons, brackets, or incorrect token ordering

PASS: tests/k0/sem_test1.kt
FAIL: tests/k0/sem_test2.kt
Expected No Error, but got:
Semantic Error at line 4: Type mismatch in assignment: cannot assign value of type float to variable of type int

Parsing completed with 1 semantic error(s)

FAIL: tests/k0/sem_test3.kt
Expected No Error, but got:
Semantic Error at line 3: Array access on a non-array type
Semantic Error at line 3: Assignment to element of immutable array
Semantic Error at line 3: Assignment to immutable variable
Semantic Error at line 3: Type mismatch in assignment: cannot assign value of type int to variable of type none
Semantic Error at line 4: Array access on a non-array type

Parsing completed with 5 semantic error(s)

PASS: tests/k0/syn_while.kt
PASS: tests/k0/syn_test1.kt
PASS: tests/kotlin/syn7.kt
PASS: tests/kotlin/lex_test42.kt
PASS: tests/kotlin/lex_test12.kt
PASS: tests/kotlin/syn2.kt
PASS: tests/kotlin/syn3.kt
PASS: tests/kotlin/lex_test53.kt
PASS: tests/kotlin/syn6.kt
PASS: tests/kotlin/lex_test39.kt
PASS: tests/kotlin/sem7.kt
PASS: tests/kotlin/lex_test19.kt
PASS: tests/kotlin/lex_test27.kt
PASS: tests/kotlin/lex_test25.kt
PASS: tests/kotlin/lex_test54.kt
PASS: tests/kotlin/syn1.kt
PASS: tests/kotlin/lex_test4.kt
PASS: tests/kotlin/lex_test11.kt
PASS: tests/kotlin/sem1.kt
PASS: tests/kotlin/lex_test21.kt
PASS: tests/kotlin/sem4.kt
PASS: tests/kotlin/lex_test13.kt
PASS: tests/kotlin/lex_test1.kt
PASS: tests/kotlin/lex_test22.kt
PASS: tests/kotlin/lex_test26.kt
PASS: tests/kotlin/lex_test24.kt
PASS: tests/kotlin/sem3.kt
PASS: tests/kotlin/lex_test51.kt
PASS: tests/kotlin/lex_test18.kt
PASS: tests/kotlin/lex_test0.kt
PASS: tests/kotlin/lex_test46.kt
PASS: tests/kotlin/lex_test36.kt
PASS: tests/kotlin/lex_test47.kt
PASS: tests/kotlin/lex_test49.kt
PASS: tests/kotlin/lex_test45.kt
PASS: tests/kotlin/lex_test20.kt
PASS: tests/kotlin/lex_test23.kt
PASS: tests/kotlin/lex_test48.kt
PASS: tests/kotlin/lex_test32.kt
PASS: tests/kotlin/lex_test37.kt
PASS: tests/kotlin/lex_test29.kt
PASS: tests/kotlin/syn4.kt
PASS: tests/kotlin/lex_test33.kt
FAIL: tests/kotlin/syn5.kt
Expected Error, but got:
Processing file: tests/kotlin/syn5.kt
DEBUG: Function 'main' recorded with 0 parameter(s), return type: any
DEBUG: Entering node: functionDeclaration (prod: 327) at line 5, scope: 0x55c57bc1a640
DEBUG: Function declaration encountered. Updating current scope.
DEBUG: Function 'main' has scope pointer: 0x55c57bc1a640
DEBUG: Entering node: main (prod: 396) at line 0, scope: 0x55c57bc1a640
DEBUG: Entering node: functionValueParameters (prod: 108) at line 1, scope: 0x55c57bc1a640
DEBUG: Entering node: block (prod: 111) at line 5, scope: 0x55c57bc1a640
DEBUG: Entering node: statements (prod: 112) at line 5, scope: 0x55c57bc1a640
DEBUG: Entering node: assignment (prod: 280) at line 3, scope: 0x55c57bc1a640
DEBUG: Propagated declaration info: identifier 'a' gets type int, mutable=0, nullable=0
DEBUG: Entering node: variableDeclaration (prod: 329) at line 3, scope: 0x55c57bc1a640
DEBUG: Entering node: a (prod: 396) at line 0, scope: 0x55c57bc1a640
DEBUG: Entering node: type (prod: 102) at line 2, scope: 0x55c57bc1a640
DEBUG: Entering node: Int (prod: 396) at line 0, scope: 0x55c57bc1a640
DEBUG: Entering node: IntegerLiteral (prod: 389) at line 2, scope: 0x55c57bc1a640
DEBUG: Entering node: 5 (prod: 389) at line 0, scope: 0x55c57bc1a640
DEBUG: Entering node: ifStatement (prod: 340) at line 5, scope: 0x55c57bc1a640
DEBUG: Entering node: comparison (prod: 298) at line 3, scope: 0x55c57bc1a640
DEBUG: Resolved Identifier 'a' to type int, mutable=0, nullable=0
DEBUG: Entering node: Identifier (prod: 396) at line 3, scope: 0x55c57bc1a640
DEBUG: Entering node: a (prod: 396) at line 0, scope: 0x55c57bc1a640
DEBUG: Entering node: IntegerLiteral (prod: 389) at line 3, scope: 0x55c57bc1a640
DEBUG: Entering node: 6 (prod: 389) at line 0, scope: 0x55c57bc1a640
DEBUG: In comparison operator at line 3. Left type: int, Right type: int
DEBUG: Entering node: functionCall (prod: 122) at line 5, scope: 0x55c57bc1a640
DEBUG: Entering node: println (prod: 396) at line 0, scope: 0x55c57bc1a640
DEBUG: Entering node: expressionList (prod: 116) at line 4, scope: 0x55c57bc1a640
DEBUG: Entering node: StringLiteral (prod: 395) at line 4, scope: 0x55c57bc1a640
DEBUG: Entering node: "less" (prod: 395) at line 0, scope: 0x55c57bc1a640
DEBUG: Checking function call for 'println' at line 0
DEBUG: Function 'println' expects 1 parameter(s), call has 1 argument(s).
No errors

PASS: tests/kotlin/lex_test16.kt
PASS: tests/kotlin/lex_test52.kt
PASS: tests/kotlin/lex_test6.kt
PASS: tests/kotlin/lex_test7.kt
PASS: tests/kotlin/lex_test40.kt
PASS: tests/kotlin/lex_test28.kt
PASS: tests/kotlin/lex_test8.kt
PASS: tests/kotlin/sem6.kt
PASS: tests/kotlin/lex_test55.kt
PASS: tests/kotlin/lex_test30.kt
PASS: tests/kotlin/sem5.kt
PASS: tests/kotlin/lex_test14.kt
PASS: tests/kotlin/lex_test31.kt
PASS: tests/kotlin/lex_test17.kt
PASS: tests/kotlin/lex_test50.kt
PASS: tests/kotlin/lex_test3.kt
PASS: tests/kotlin/lex_test43.kt
PASS: tests/kotlin/lex_test44.kt
PASS: tests/kotlin/lex_test9.kt
PASS: tests/kotlin/lex_test2.kt
PASS: tests/kotlin/lex_test35.kt
PASS: tests/kotlin/lex_test34.kt
FAIL: tests/kotlin/sem2.kt
Expected Error, but got:
Processing file: tests/kotlin/sem2.kt
DEBUG: Function 'fun1' recorded with 0 parameter(s), return type: any
DEBUG: Function 'fun2' recorded with 0 parameter(s), return type: any
DEBUG: Function 'main' recorded with 0 parameter(s), return type: any
DEBUG: Entering node: topLevelObjectList (prod: 101) at line 9, scope: 0x5603d999a8e0
DEBUG: Entering node: functionDeclaration (prod: 327) at line 6, scope: 0x5603d999a8e0
DEBUG: Function declaration encountered. Updating current scope.
DEBUG: Function 'fun1' has scope pointer: 0x5603d999a2e0
DEBUG: Entering node: fun1 (prod: 396) at line 0, scope: 0x5603d999a2e0
DEBUG: Entering node: functionValueParameters (prod: 108) at line 1, scope: 0x5603d999a2e0
DEBUG: Entering node: block (prod: 111) at line 6, scope: 0x5603d999a2e0
DEBUG: Entering node: statements (prod: 112) at line 6, scope: 0x5603d999a2e0
DEBUG: Entering node: functionDeclaration (prod: 327) at line 4, scope: 0x5603d999a2e0
DEBUG: Function declaration encountered. Updating current scope.
DEBUG: Function 'fun2' has scope pointer: 0x5603d999a5e0
DEBUG: Entering node: fun2 (prod: 396) at line 0, scope: 0x5603d999a5e0
DEBUG: Entering node: functionValueParameters (prod: 108) at line 2, scope: 0x5603d999a5e0
DEBUG: Entering node: block (prod: 111) at line 4, scope: 0x5603d999a5e0
DEBUG: Entering node: functionCall (prod: 122) at line 4, scope: 0x5603d999a5e0
DEBUG: Entering node: println (prod: 396) at line 0, scope: 0x5603d999a5e0
DEBUG: Entering node: expressionList (prod: 116) at line 3, scope: 0x5603d999a5e0
DEBUG: Entering node: StringLiteral (prod: 395) at line 3, scope: 0x5603d999a5e0
DEBUG: Entering node: "nested function" (prod: 395) at line 0, scope: 0x5603d999a5e0
DEBUG: Checking function call for 'println' at line 0
DEBUG: Function 'println' expects 1 parameter(s), call has 1 argument(s).
DEBUG: Entering node: functionCall (prod: 122) at line 6, scope: 0x5603d999a2e0
DEBUG: Entering node: fun2 (prod: 396) at line 0, scope: 0x5603d999a2e0
DEBUG: Checking function call for 'fun2' at line 0
DEBUG: Function 'fun2' expects 0 parameter(s), call has 0 argument(s).
DEBUG: Entering node: functionDeclaration (prod: 327) at line 9, scope: 0x5603d999a8e0
DEBUG: Function declaration encountered. Updating current scope.
DEBUG: Function 'main' has scope pointer: 0x5603d999a8e0
DEBUG: Entering node: main (prod: 396) at line 0, scope: 0x5603d999a8e0
DEBUG: Entering node: functionValueParameters (prod: 108) at line 7, scope: 0x5603d999a8e0
DEBUG: Entering node: block (prod: 111) at line 9, scope: 0x5603d999a8e0
DEBUG: Entering node: functionCall (prod: 122) at line 9, scope: 0x5603d999a8e0
DEBUG: Entering node: fun1 (prod: 396) at line 0, scope: 0x5603d999a8e0
DEBUG: Checking function call for 'fun1' at line 0
DEBUG: Function 'fun1' expects 0 parameter(s), call has 0 argument(s).
No errors

PASS: tests/kotlin/lex_test38.kt
PASS: tests/kotlin/lex_test10.kt
PASS: tests/kotlin/lex_test41.kt
PASS: tests/kotlin/lex_test15.kt
PASS: tests/kotlin/lex_test5.kt

=== TEST SUMMARY ===
Total Tests: 157
Passed: 150
Failed: 7
